Mytinerary/View Controllers/InputEventViewController.m

```diff
@@ -0,0 +1,294 @@
+        if (succeeded) {
+            NSLog(@"food event successfully initialized!");
+            [self dismissViewControllerAnimated:YES completion:nil];
+        }
+        else {
+            NSLog(@"error initializing food event: %@", error);
+        }
+    }];
+}
+
+- (void)initHotelEvent {
+    float cost = (float)0.0;
+    if (![self.eventInputHotelView.costTextField.text isEqualToString:@""]) {
+        cost = [self.eventInputHotelView.costTextField.text floatValue];
+    }
+    
+    int selectedHotelTypeIdx = (int)[self.eventInputHotelView.typePickerView selectedRowInComponent:0];
+    NSString *selectedHotelType = self.hotelTypePickerData[selectedHotelTypeIdx];
+    
+    [Event initHotelEvent:self.eventInputSharedView.titleTextField.text eventDescription:self.eventInputSharedView.descriptionTextView.text address:self.eventInputHotelView.locationTextField.text startTime:self.eventInputSharedView.startTimeDatePicker.date endTime:self.eventInputSharedView.endTimeDatePicker.date hotelType:selectedHotelType cost:cost notes:self.eventInputHotelView.notesTextView.text withCompletion:^(BOOL succeeded, NSError * _Nullable error) {
+        if (succeeded) {
+            NSLog(@"hotel event successfully initialized!");
+            [self dismissViewControllerAnimated:YES completion:nil];
+        }
+        else {
+            NSLog(@"error initializing hotel event: %@", error);
+        }
+    }];
+}
+
+/*
+#pragma mark - Navigation
+
+// In a storyboard-based application, you will often want to do a little preparation before navigation
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    // Get the new view controller using [segue destinationViewController].
+    // Pass the selected object to the new view controller.
+}
+*/
+
+
+// picker view functions
+- (NSInteger)numberOfComponentsInPickerView:(nonnull UIPickerView *)pickerView {
+    return 1;
+}
+
+- (NSInteger)pickerView:(nonnull UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component {
+    
+    switch (pickerView.tag) {
+        case 0:
+            return self.eventCategoryPickerData.count;
+            break;
+        case 1:
+            return self.transportationTypePickerData.count;
+            break;
+        case 2:
+            return self.foodCostPickerData.count;
+            break;
+        case 3:
+            return self.hotelTypePickerData.count;
+            break;
+        default:
+            return 1;
+            break;
+    }
+}
+
+- (NSString *)pickerView:(UIPickerView *)pickerView titleForRow:(NSInteger)row forComponent:(NSInteger)component {
+    
+    switch (pickerView.tag) {
+        case 0:
+            return self.eventCategoryPickerData[row];
+            break;
```

Comment:
I don't think you need "return" and "break" in a switch case, just one of the two.